script/pathfinder.lua

```diff
@@ -0,0 +1,293 @@
+            if clean_path_steps_enabled then
+                path = pathfinder.clean_path_steps(path, 2.5)
+            end
+            if request.job then
+                global.job_bundles[request.job.bundle_index][1]["path_active"] = true
+            end
+            pathfinder.set_autopilot(request.unit, path)
+        end
+    end
+    global.pathfinder_requests[event.id] = nil
+end
+
+-------------------------------------------------------------------------------
+--  Utility
+-------------------------------------------------------------------------------
+
+---@param unit LuaEntity
+---@param path PathfinderWaypoint[]
+function pathfinder.set_autopilot(unit, path) -- set path
+    if unit and unit.valid then
+        unit.autopilot_destination = nil
+        for i, waypoint in ipairs(path) do
+            unit.add_autopilot_destination(waypoint.position)
+        end
+    end
+end
+
+---@param surface LuaSurface
+---@param position MapPosition
+---@return MapPosition?
+function pathfinder.find_non_colliding_position(surface, position, job) -- find a position to stand / walk to
+    local new_position
+    local bb
+    local params = {
+        {size = 12, radius = 64},
+        {size = 5, radius = 16},
+        {size = 5, radius = 8},
+        {size = 1, radius = 12},
+        {size = 1, radius = 5}
+        }
+    for _, param in pairs(params) do
+        new_position = surface.find_non_colliding_position("constructron_pathing_proxy_" .. param.size, position, param.radius, non_colliding_position_accuracy, false)
+        if new_position then
+            bb = param
+        end
+    end
+    if new_position then
+        if job and (job.action == "go_to_position") then -- update the job for condition check
+            global.job_bundles[job.bundle_index][1]["leave_args"][1] = new_position
+        end
+        -- game.print('bounding_box:'.. serpent.block(bb) ..'')
+        return new_position -- return for the new request
+    end
+end
+
+function pathfinder.check_pathfinder_requests_timeout()
+    for key, request in pairs(global.pathfinder_requests) do
+        if ((game.tick - request.request_tick) > 60 * 60) then -- one minute
+            global.pathfinder_requests[key] = nil
+        end
+    end
+end
+
+---@param path PathfinderWaypoint[]
+---@return PathfinderWaypoint[]
+function pathfinder.clean_linear_path(path) -- removes points on the same line except the start and the end.
+    local new_path = {}
+    for i, waypoint in ipairs(path) do
+        if i >= 2 and i < #path then
+            local prev_angle = math.atan2(waypoint.position.y - path[i - 1].position.y, waypoint.position.x - path[i - 1].position.x)
+            local next_angle = math.atan2(path[i + 1].position.y - waypoint.position.y, path[i + 1].position.x - waypoint.position.x)
+            if math.abs(prev_angle - next_angle) > 0.005 then
```

Comment:
maybe the magic number `0.005` should be a variable (maybe named `clean_linear_path_max_angle_diff`?) at the top of the file (with a accommodating comment mentioning that this is the max angle difference in radians for 2 segments to be merged into 1 segment).